### PR TITLE
feat(appconnect): New table to track when sentry last checked for new builds on App Store Connect

### PR DESCRIPTION
**Pull request by relaxolotl**
Does what it says on the tin. This is meant to replace a project option that currently tracks the same info, which can be found by searching for `APPSTORECONNECT_BUILD_REFRESHES_OPTION`.

### CODE REVIEW

File: src/sentry/models/latestappconnectbuildscheck.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    specific appconnect source in a project.
+    """
+
+    __include_in_export__ = False
+
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False)
+
+    # The symbol source's unique identifier in the project. Also known as the id on an
+    # AppStoreConnectConfig.
+    source_id = models.CharField(max_length=200, default="")
```
Comment (flub) [inline src/sentry/models/latestappconnectbuildscheck.py:25]: Providing a default here seems dangerous?  Can we not require that you provide this or you can not create a row?  If there something in the database that stops us from doing that this is a reasonable compromise and you can ignore this comment though.

File: src/sentry/models/latestappconnectbuildscheck.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    # When sentry last checked App Store Connect.
+    last_fetched = models.DateTimeField(default=timezone.now)
+
+    @classmethod
+    def refresh_date(cls, project, source_id):
```
Comment (flub) [inline src/sentry/models/latestappconnectbuildscheck.py:31]: rather low priority nit: i find "refresh" somewhat confusing and would choose "update" myself.  No refresh action is being done, a row is being updated.  🤷

File: src/sentry/models/latestappconnectbuildscheck.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                project=project,
+                source_id=source_id,
+                last_fetched=timezone.now(),
+            )
+            latest_check.save()
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_latestappconnectbuildscheck"
+        unique_together = (("project", "source_id"),)
```
Comment (flub) [inline src/sentry/models/latestappconnectbuildscheck.py:47]: another nit, if you make the outer a list you don't have to muck about with a trailing comma: `[("project", "source_id")]`

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
Another thing you should definitely still do: add it to https://github.com/getsentry/sentry/blob/551f77a7782d62f8d7a32c19c52b03bfcc663706/src/sentry/deletions/defaults/project.py#L17 so that it gets deleted when the project is deleted.  I don't think we have a way to delete entries as symbol sources are removed, but that's pretty extreme to become a problem, but lets at least attempt to delete it when the project is deleted.

File: src/sentry/models/latestappconnectbuildscheck.py (hunk 1 above)
Comment (relaxolotl) [reply src/sentry/models/latestappconnectbuildscheck.py:25]: strange, i bumped into an error while creating/applying the migration that yelled about defaults for this field but i can't seem to reproduce it anymore. i'll remove this, thanks!

**Comment by relaxolotl** [comment]
cc @getsentry/owners-migrations when somebody has the time could I bug one of you to take a look at the changes here?

File: src/sentry/models/latestappconnectbuildscheck.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    last_checked = models.DateTimeField(default=timezone.now)
+
+    @classmethod
+    def update_date(cls, project, source_id, new_date):
+        if new_date is None:
+            new_date = timezone.now()
+        try:
+            latest_check = cls.objects.get(project=project, source_id=source_id)
+            latest_check.last_checked = new_date
+            latest_check.save(update_fields=["last_checked"])
```
Comment (wedamija) [inline src/sentry/models/latestappconnectbuildscheck.py:38]: jfyi, we have a `.update`  you can use to do this - `latest_check.update(last_checked=new_date)`.

File: src/sentry/models/latestappconnectbuildscheck.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False)
+
+    # The symbol source's unique identifier in the project. Also known as the id on an
+    # AppStoreConnectConfig.
+    source_id = models.CharField(max_length=200)
+
+    # When sentry last checked App Store Connect for new builds. A check may return zero new builds,
+    # or no builds at all if all existing builds have expired.
+    last_checked = models.DateTimeField(default=timezone.now)
+
```
Comment (wedamija) [inline src/sentry/models/latestappconnectbuildscheck.py:30]: I'd recommend you add a `date_added` field. It just tends to be useful for debugging. If you subclass `DefaultFieldsModel` it'll add `date_added` and `date_updated` for you automatically, or if you prefer to not have `date_updated` you can just add the field separately.

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/latestappconnectbuildscheck.py (hunk 5 above)
Comment (wedamija) [reply src/sentry/models/latestappconnectbuildscheck.py:30]: (You'll need to regenerate your migration after adding the default model)

Comment (relaxolotl) [reply src/sentry/models/latestappconnectbuildscheck.py:30]: oh dang it i was hoping i could skip that, thanks!

**Review by wedamija — APPROVED** [review]

**Pull request merged**